src/Orchard.Web/Modules/Orchard.DynamicForms/Drivers/TaxonomyElementDriver.cs

```diff
@@ -133,6 +137,24 @@ public TaxonomyElementDriver(IFormsBasedElementServices formsServices, ITaxonomy
             context.ElementShape.Metadata.Alternates.Add(String.Format("Elements_{0}_{1}__{2}", typeName, displayType, element.InputType));
         }
 
+        protected override void OnExporting(Taxonomy element, ExportElementContext context) {
+            var taxonomy = element.TaxonomyId != null ? _contentManager.Get<TaxonomyPart>(element.TaxonomyId.Value) : default(TaxonomyPart);
+            var taxonomyIdentity = taxonomy != null ? _contentManager.GetItemMetadata(taxonomy).Identity.ToString() : default(string);
+
+            if (taxonomyIdentity != null)
+                context.ExportableData["TaxonomyId"] = taxonomyIdentity;
+        }
+
+        protected override void OnImportCompleted(Taxonomy element, ImportElementContext context) {
```

Comment:
why not `OnImporting` ?

Reply:
ProjectElementDriver does the same, so this PR applies that logic completely to this case.